File: isletimsistemlericalismasiJava-calismalarim/app/src/main/java/com/example/isletimsistemicalismasi/uygulamalar/rehber/entity/RehberDao.java
package com.example.isletimsistemicalismasi.uygulamalar.rehber.entity;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.example.isletimsistemicalismasi.uygulamalar.rehber.model.RehberKisiler;

import java.util.ArrayList;

public class RehberDao {
    RehberVeritabaniYardimcisi vt;

    public RehberDao(RehberVeritabaniYardimcisi vt) {
        this.vt = vt;
    }
    public void kisiEkle(String kisi_isim,String kisi_numara){
        SQLiteDatabase dbx = vt.getWritableDatabase();
        ContentValues degerler = new ContentValues();
        degerler.put("kisi_isim",kisi_isim);
        degerler.put("kisi_numara",kisi_numara);
        dbx.insertOrThrow("rehber",null,degerler);
    }
    public ArrayList<RehberKisiler> tumKisiler(){
        ArrayList<RehberKisiler> tumKisiler = new ArrayList<>();
        SQLiteDatabase dbx = vt.getWritableDatabase();
        Cursor c = dbx.rawQuery("SELECT * FROM rehber",null);
        while (c.moveToNext()){
            @SuppressLint("Range") RehberKisiler kisi = new RehberKisiler(c.getInt(c.getColumnIndex("kisi_id")),
                    c.getString(c.getColumnIndex("kisi_isim"))
                    ,c.getString(c.getColumnIndex("kisi_numara")));
            tumKisiler.add(kisi);
        }
        dbx.close();
        return tumKisiler;
    }
    public void kisiSil(int kisi_id){
        SQLiteDatabase dbx = vt.getWritableDatabase();
        dbx.delete("rehber","kisi_id=?",new String[]{String.valueOf(kisi_id)});
        dbx.close();
    }
    public void kisiGuncelle(int kisi_id,String kisi_isim,String kisi_numara){
        SQLiteDatabase dbx = vt.getWritableDatabase();
        ContentValues degerler = new ContentValues();
        degerler.put("kisi_id",kisi_id);
        degerler.put("kisi_isim",kisi_isim);
        degerler.put("kisi_numara",kisi_numara);
        dbx.update("rehber",degerler,"kisi_id=?",new String[]{String.valueOf(kisi_id)});
        dbx.close();
    }

}

File: isletimsistemlericalismasikotlin-calismalarim/app/src/main/java/com/example/isletimsistemlericalismasikotlin/uygulamalar/rehber/entity/RehberDao.kt
package com.example.isletimsistemlericalismasikotlin.uygulamalar.rehber.entity

import android.content.ContentValues
import com.example.isletimsistemlericalismasikotlin.uygulamalar.rehber.model.RehberKisiler

class RehberDao(private val vt: RehberVeritabaniYardimcisi) {

    fun kisiEkle(kisiIsim: String, kisiNumara: String) {
        val dbx = vt.writableDatabase
        val degerler = ContentValues().apply {
            put("kisi_isim", kisiIsim)
            put("kisi_numara", kisiNumara)
        }
        dbx.insertOrThrow("rehber", null, degerler)
        dbx.close()
    }

    fun tumKisiler(): ArrayList<RehberKisiler> {
        val tumKisiler = ArrayList<RehberKisiler>()
        val dbx = vt.writableDatabase
        val cursor = dbx.rawQuery("SELECT * FROM rehber", null)

        while (cursor.moveToNext()) {
            val kisi = RehberKisiler(
                cursor.getInt(cursor.getColumnIndexOrThrow("kisi_id")),
                cursor.getString(cursor.getColumnIndexOrThrow("kisi_isim")),
                cursor.getString(cursor.getColumnIndexOrThrow("kisi_numara"))
            )
            tumKisiler.add(kisi)
        }
        cursor.close()
        dbx.close()
        return tumKisiler
    }

    fun kisiSil(kisiId: Int) {
        val dbx = vt.writableDatabase
        dbx.delete("rehber", "kisi_id=?", arrayOf(kisiId.toString()))
        dbx.close()
    }

    fun kisiGuncelle(kisiId: Int, kisiIsim: String, kisiNumara: String) {
        val dbx = vt.writableDatabase
        val degerler = ContentValues().apply {
            put("kisi_id", kisiId)
            put("kisi_isim", kisiIsim)
            put("kisi_numara", kisiNumara)
        }
        dbx.update("rehber", degerler, "kisi_id=?", arrayOf(kisiId.toString()))
        dbx.close()
    }
}
